pymove/tests/core/test_data_frame.py

```diff
@@ -0,0 +1,1102 @@
+import os
+from datetime import date
+
+from dask.dataframe import DataFrame as DaskDataFrame
+from numpy import nan, ndarray
+from numpy.testing import assert_allclose, assert_array_equal
+from pandas import DataFrame, Series, Timedelta, Timestamp
+from pandas.testing import assert_frame_equal, assert_series_equal
+
+import pymove
+from pymove import DaskMoveDataFrame, MoveDataFrame, PandasMoveDataFrame, read_csv
+from pymove.utils.constants import (
+    DATE,
+    DATETIME,
+    DAY,
+    DIST_PREV_TO_NEXT,
+    DIST_TO_PREV,
+    HOUR,
+    HOUR_SIN,
+    LATITUDE,
+    LONGITUDE,
+    PERIOD,
+    SITUATION,
+    SPEED_PREV_TO_NEXT,
+    TID,
+    TIME_PREV_TO_NEXT,
+    TRAJ_ID,
+    TYPE_DASK,
+    TYPE_PANDAS,
+    UID,
+    WEEK_END,
+)
+
+list_data = [
+    [39.984094, 116.319236, '2008-10-23 05:53:05', 1],
+    [39.984198, 116.319322, '2008-10-23 05:53:06', 1],
+    [39.984224, 116.319402, '2008-10-23 05:53:11', 2],
+    [39.984224, 116.319402, '2008-10-23 05:53:11', 2]
+]
+
+str_data_default = """
+lat,lon,datetime,id
+39.984093,116.319236,2008-10-23 05:53:05,4
+39.9842,116.319322,2008-10-23 05:53:06,1
+39.984222,116.319402,2008-10-23 05:53:11,2
+39.984222,116.319402,2008-10-23 05:53:11,2
+"""
+
+str_data_different = """
+latitude,longitude,time,traj_id
+39.984093,116.319236,2008-10-23 05:53:05,4
+39.9842,116.319322,2008-10-23 05:53:06,1
+39.984222,116.319402,2008-10-23 05:53:11,2
+39.984222,116.319402,2008-10-23 05:53:11,2
+"""
+
+str_data_missing = """
+39.984093,116.319236,2008-10-23 05:53:05,4
+39.9842,116.319322,2008-10-23 05:53:06,1
+39.984222,116.319402,2008-10-23 05:53:11,2
+39.984222,116.319402,2008-10-23 05:53:11,2
+"""
+
+
+def _default_move_df():
+    return MoveDataFrame(
+        data=list_data, latitude=LATITUDE, longitude=LONGITUDE,
+        datetime=DATETIME, traj_id=TRAJ_ID
+    )
+
+
+def _default_pandas_df():
+    return DataFrame(data=[
+        [39.984094, 116.319236, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984198, 116.319322, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984224, 116.319402, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=['lat', 'lon', 'datetime', 'id'], index=[0, 1, 2, 3]
+    )
+
+
+def _has_columns(data):
+    if LATITUDE in data and LONGITUDE in data and DATETIME in data:
+        return True
+    return False
+
+
+def _validate_move_data_frame_data(data):
+    try:
+        if data.dtypes.lat != 'float64':
+            return False
+        if data.dtypes.lon != 'float64':
+            return False
+        if data.dtypes.datetime != 'datetime64[ns]':
+            return False
+        return True
+    except AttributeError:
+        print(AttributeError)
+
+
+def test_move_data_frame_from_list():
+    move_df = _default_move_df()
+    assert _has_columns(move_df)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)